### ISSUE TITLE
IPFS Companion not switching to local Gateway even when X-Ipfs-Path is provided

### ISSUE THREAD

**Issue by provirus** [ipfs-shipyard/ipfs-companion#953]
# Browser Setup

- OS: Linux Ubuntu 20.04.1 LTS
- Browser: Chrome Version 87.0.4280.88 (Official Build) (64-bit)
- IPFS Companion Version: 2.16.0
- IPFS Companion has been reset to default preferences

# Working for main site

Steps:
- Going on https://ipfs.io/
- [OK] URL becomes http://ipfs.io.ipns.localhost:8080/

# Issue with a custom site

## Steps to repreduce

Steps:
- Going on http://test.cloud.foilen.com/
- [ERROR] URL is stuck to http://test.cloud.foilen.com/
    - Expected to become http://test.cloud.foilen.com.ipns.localhost:8080/ or http://localhost:8080/ipns/test.cloud.foilen.com

## Custom site setup

[OK] DNSLink

```
dig test.cloud.foilen.com TXT

    ;; ANSWER SECTION:
    test.cloud.foilen.com.	1	IN	TXT	"dnslink=/ipns/k51qzi5uqu5dhuj92m8egzrbx6e0apodpebvs7y4fqe1rc6rxb1hsiwgovl94o"

```

[OK] X-Ipfs-Path Header

```
curl -v https://test.cloud.foilen.com 2>&1 | grep X-Ipfs-Path
    < X-Ipfs-Path: /ipns/test.cloud.foilen.com/
```

## Misc

Also, using my local Gateway directly is working

- [OK] Going on http://test.cloud.foilen.com.ipns.localhost:8080/

--------

Could you let me know if my setup is missing something or fix the Chrome extension if there is a bug with it?

thanks

**Comment by jessicaschilling** [comment]
Thank you for the detailed report. Many of the core team are at reduced capacity for the holidays, so it's suggested to open this as a post in the forums at discuss.ipfs.io so your question can be more easily seen by the community as a whole. Thank you!

**Comment by lidel** [comment]
Hi @simon, thank you for bringing this issue up.
I can confirm this is a bug related to parsing  `/ipns/` in DNSLink records, we are tracking it in: https://github.com/ipfs/go-ipfs/issues/6454#issuecomment-739055758

**TLDR workaround:** until it is fixed upstream in go-ipfs, you need to use immutable paths starting with `/ipfs/{cid}/`

**Comment by provirus** [comment]
I confirm the workaround works. Thanks

**Comment by lidel** [comment]
I believe this is fixed in [v2.17.1](https://github.com/ipfs-shipyard/ipfs-companion/releases/tag/v2.17.1)  :-)
